services/ad-tech/src/public/js/ssp/ssp-tag.js

```diff
@@ -17,54 +20,200 @@
  *   setup. This is included on certain tagged publisher pages.
  *
  * What does this script do:
- *   This script injects an iframe in its own origin to invoke runAdAuction()
- *   and renders the ad in a fenced frame.
+ *   This script orchestrates the loading of other seller modules to deliver
+ *   relevant ads for all the ad slots available on the publisher page. This
+ *   script reads the ad slot configurations set by the publisher and injects
+ *   an iframe for each ad slot. These iframes then individually execute ad
+ *   auctions to select the ad as per publisher configurations.
  */
-(async () => {
+(() => {
+  /** Current script domain name. */
+  const CURR_HOSTNAME = new URL(document.currentScript.src).hostname;
+  // ********************************************************
+  // HELPER FUNCTIONS
+  // ********************************************************
+  /** Logs to console. */
+  const log = (message, context) => {
+    console.log('[PSDemo] Seller', CURR_HOSTNAME, 'ad tag', message, {context});
+  };
+
+  /** Returns frame URL with page context as search query. */
+  const getIframeUrlWithPageContext = (pathname) => {
+    if (!pathname) {
+      return;
+    }
+    // Construct iframe URL using current script origin.
+    const $script = document.currentScript;
+    const src = new URL($script.src);
+    src.pathname = pathname;
+    // Append query parameters from script dataset context.
+    for (const datakey in $script.dataset) {
+      src.searchParams.append(datakey, $script.dataset[datakey]);
+    }
+    // Append query params from page URL.
+    const currentUrl = new URL(location.href);
+    for (const [key, value] of currentUrl.searchParams) {
+      src.searchParams.append(key, value);
+    }
+    src.searchParams.append('publisher', location.origin);
+    src.searchParams.append('title', document.title);
+    return src;
+  };
+
+  /** Injects an iframe using the current script's reference. */
+  const injectAndReturnIframe = ({src, divId, size, attributes}) => {
+    if (!src) {
+      return;
+    }
+    const iframeEl = document.createElement('iframe');
+    iframeEl.src = src;
+    [iframeEl.width, iframeEl.height] = size;
+    if (attributes && 'object' === typeof attributes) {
+      for (const [key, value] of Object.entries(attributes)) {
+        iframeEl.setAttribute(key, value);
+      }
+    }
+    log('injecting iframe', {src, divId, iframeEl});
+    document.getElementById(divId).appendChild(iframeEl);
+    return iframeEl;
+  };
+
   /** Listen for potential post messages from DSPs. */
   const addEventListenerForDspPostMessages = () => {
-    window.addEventListener('message', (event) => {
+    window.addEventListener('message', async (event) => {
       if (!event.origin.startsWith('https://privacy-sandbox-demos-dsp')) {
```

Comment:
Remove hardcoded url. replace by DSP_HOST

Reply:
I see this was fixed, resolving this comment